### PR TITLE
Add "tag" and "untag" command (tags) to tag/untag an item

### LINKED ISSUE #115: Add "tag" command (tags) to tag an item

### LINKED ISSUE #116: Add "untag" command (tags) to untag an item

### PR DESCRIPTION
**Pull request by Eugenetanwl3881**
Closes #116 
Closes #115 

Tags and Untags an item if tag provided and item index provided is valid.
Implements the logic behind tagging/untagging of item.

 UI to show the tagging and untagging is done in PR #171. Please merge this first.

Outstanding tasks:

- Writing Test Cases

### CODE REVIEW

File: src/main/java/seedu/foodrem/logic/parser/TagCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        String name = ParserUtil.parseTagName(argMultimap.getValue(PREFIX_TAG_NAME).get()).toString();
+
+        return new TagCommand(name, index);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```
Comment (Ferusel) [inline src/main/java/seedu/foodrem/logic/parser/TagCommandParser.java:42]: I believe this has already been defined in AddCommandParser, shall we move this to e.g. ParserUtil?

File: src/main/java/seedu/foodrem/logic/parser/UntagCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        String name = ParserUtil.parseTagName(argMultimap.getValue(PREFIX_TAG_NAME).get()).toString();
+
+        return new UntagCommand(name, index);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```
Comment (Ferusel) [inline src/main/java/seedu/foodrem/logic/parser/UntagCommandParser.java:43]: Ditto

File: src/main/java/seedu/foodrem/logic/commands/tagcommands/UntagCommand.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    /**
+     * Creates and returns a {@code Item} with the tagSet of {@code itemToUntag}
+     * edited
+     */
+    private static Item createUntaggedItem(Item itemToUntag, Tag tag) {
+        assert itemToUntag != null;
+
+        itemToUntag.removeTag(tag);
```
Comment (Ferusel) [inline src/main/java/seedu/foodrem/logic/commands/tagcommands/UntagCommand.java:58]: Question regarding the implementation: I see that removeTag removes the existence of that tag from FoodRem "Removes {@code key} from this {@code AddressBook}". Will calling this cause bugs where all references to that tag will be removed?

File: src/main/java/seedu/foodrem/logic/parser/TagCommandParser.java (hunk 1 above)
Comment (Eugenetanwl3881) [reply src/main/java/seedu/foodrem/logic/parser/TagCommandParser.java:42]: Yes, good catch. I will refactor this into ParserUtil for AddCommandParser and TagCommandParser to use.

File: src/main/java/seedu/foodrem/logic/commands/tagcommands/UntagCommand.java (hunk 3 above)
Comment (Eugenetanwl3881) [reply src/main/java/seedu/foodrem/logic/commands/tagcommands/UntagCommand.java:58]: This method call to removeTag refers to the method implemented in the Item class (not the FoodRem class). But yes, I think it's better to rename the method to removeItemTag instead, for clarity.

**Review by Ferusel — APPROVED** [review]
LGTM!

**Pull request merged**